### PR TITLE
default field selectors

### PR DESCRIPTION
**Pull request by deads2k**
We have a lot of code around field selectors that doesn't add much value. Every gettable resources probably wants name and namespace by default and since they all use metav1 (today), we can assign that as a default. If we think we'll always have metav1 style name and namespace, then this makes a reasonable default and you can always set something different.

This removes cruft and avoids the risk of accidentally forgetting a field selector.  

@kubernetes/sig-api-machinery-misc @smarterclayton

### CODE REVIEW

**Comment by enj** [comment]
LGTM

I do enjoy PRs that delete boilerplate 😄

File: staging/src/k8s.io/apimachinery/pkg/runtime/conversion.go [hunk 1]
```diff
@@ -19,13 +19,28 @@ limitations under the License.
...
+// A cluster scoped resource specifying namespace empty works fine and specifying a particular
+// namespace will return no results, as expected.
+func DefaultMetaV1FieldSelectorConversion(label, value string) (string, string, error) {
+	switch label {
+	case "metadata.name":
+		return label, value, nil
+	case "metadata.namespace":
+		return label, value, nil
+	default:
+		return "", "", fmt.Errorf("%q is not a known field selector: only %q, %q", label, "metadata.name", "metadata.namespace")
```
Comment (lavalamp) [inline staging/src/k8s.io/apimachinery/pkg/runtime/conversion.go]: message is a sentence fragment

**Comment by lavalamp** [comment]
/lgtm
/approve

Fix that message if you get a chance though, thanks.

**Pull request merged**